### ISSUE TITLE
Download Progress Bar Breaks Download Functionality

### ISSUE THREAD

**Issue by alvin-nt** [lukesampson/scoop#1334]
Seems like the new download bar is breaking some download. This occurs on stuff that requires fetching resources that do not display its file size. Perhaps related to https://github.com/lukesampson/scoop/pull/1329?

## Environment
* Windows 10 64-bit v1607
* PowerShell 5.1

## Steps to reproduce
1. Have a clean Windows install
2. Install scoop using the method described in http://scoop.sh/
3. Run `scoop install pshazz` or `scoop install concfg`
4. The PS terminal is stuck on infinite loop, trying to produce the 'correct' download bar.

## Affected apps
1. `pshazz`
2. `concfg`

**Comment by rrelmy** [comment]
@deevus I can reproduce it, can you take a look at it?

The download URLs for github have a redirect, maybe that's a hint

**Comment by deevus** [comment]
Will have to wait until later unless someone else can give this a go

**Comment by alvin-nt** [comment]
Is it currently possible for the user to disable the download bar through `scoop config`?

**Comment by lukesampson** [comment]
This looks like it might be a concurrency issue, where for small downloads `ContentLength` can be read from `HttpWebRequest` before it has been set, and the default of -1 causes the infinite loop.

As a temporary workaround, I've disabled progress when `ContentLength < 0`, which should keep installs working until we can address this properly.

To fix it properly, we probably just need to ensure content has been received before reading the ContentLength property. This was a problem with the original code before the new progress bar was added, it just glitched rather than entering an infinite loop.
